modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/K6ClientCodegen.java

```diff
@@ -744,9 +886,277 @@ private static String getAccept(OpenAPI openAPI, Operation operation) {
 
         return accepts;
     }
-    
+
     @Override
     protected ImmutableMap.Builder<String, Lambda> addMustacheLambdas() {
         return super.addMustacheLambdas().put("handleParamValue", new ParameterValueLambda());
     }
+
+    /**
+     * We're iterating over paths but grouping requests across paths, therefore
+     * these need to be aggregated.
+     *
+     * @param requestGroups
+     * @param groupName
+     * @param variables
+     * @param requests
+     */
+    private void addOrUpdateRequestGroup(Map<String, HTTPRequestGroup> requestGroups, String groupName,
+            Set<Parameter> variables, Map<Integer, HTTPRequest> requests) {
+        if (requestGroups.containsKey(groupName)) {
+            HTTPRequestGroup existingHTTPRequestGroup = requestGroups.get(groupName);
+            existingHTTPRequestGroup.addRequests(requests);
+            existingHTTPRequestGroup.addVariables(variables);
+        } else {
+            requestGroups.put(groupName, new HTTPRequestGroup(groupName, variables, requests));
+        }
+    }
+
+    /**
+     * If `X_OPERATION_DATAEXTRACT` K6 vendor extension is present, extract info
+     * from it.
+     *
+     * @param openAPI
+     * @return
+     */
+    private Map<String, DataExtractSubstituteParameter> getDataExtractSubstituteParameters(OpenAPI openAPI) {
+        Map<String, DataExtractSubstituteParameter> dataExtractSubstituteParams = new HashMap<>();
+
+        for (String path : openAPI.getPaths().keySet()) {
+            for (Map.Entry<PathItem.HttpMethod, Operation> methodOperation : openAPI.getPaths().get(path)
+                    .readOperationsMap().entrySet()) {
+
+                final PathItem.HttpMethod method = methodOperation.getKey();
+                final Operation operation = methodOperation.getValue();
+                final CodegenOperation cgOperation = super.fromOperation(path, method.name(), operation, null);
+
+                if (cgOperation.getHasVendorExtensions()
+                        && cgOperation.vendorExtensions.containsKey(X_OPERATION_DATAEXTRACT)
+                        && cgOperation.vendorExtensions.get(X_OPERATION_DATAEXTRACT) instanceof java.util.Map) {
```

Comment:
`instanceof java.util.Map`, don't we have `Map` already imported to do `intanceof Map`?

Reply:
That's a boolean check.